backstop_data/engine_scripts/puppet/price-set/preview-all-fields-modal.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = async (engine, scenario, viewport) => {
+  const priceFieldsEmpty = await engine.$('#crm-main-content-wrapper .messages');
+
+  await require('./view-and-edit-price-fields')(engine, scenario, viewport);
+  if(priceFieldsEmpty) {
```

Comment:
We have so much repeating blocks here. Is it possible to create a function somewhere, like this:

```js
// pseudocode
promise checkIfPriceFieldsAreEmpty ()
  if(priceFieldsEmpty) {
    console.warn('No Price list present!')
    console.log('Taking the "Price Fields List" Page screenshot..');

    return reject()
```

So you can use this func and will not need to duplicate the console.log block of code:

```js
checkIfPriceFieldsAreEmpty().then(() => {
  await engine.click('a[href*="/civicrm/admin/price?action=preview"]');
  await engine.waitForSelector('.CRM_Price_Form_Preview');
});
```

Something like that.

Reply:
Good Point! Let me do it.